modules/exploits/multi/http/splunk_upload_app_exec.rb

```diff
@@ -306,10 +448,27 @@ def do_get_csrf(uri)
         end
       end
     end
-
     fail_with(Failure::Unknown, "csrf form Key not found") unless @csrf_form_key
   end
 
+  # version  7.2.x only
+  def do_get_state_token(uri)
+     print_status("Fetching state token from #{uri}")
+     res = send_request_cgi(
+       'uri'    => uri,
+       'method' => 'GET',
+       'cookie' => @auth_cookies
+     )
+     #puts res
+     res.body.match('name=\"state\" value="(.*)"') # Version 5
+     @state_token = Regexp.last_match(1)
+
+   unless @state_token
+     fail_with(Failure::Unknown, "state token form Key not found") unless @state_token
```

Comment:
Duplicated check. Remove.

```suggestion
     fail_with(Failure::Unknown, "state token form Key not found")
```